imageBuilder.go

```diff
@@ -3,20 +3,67 @@ package main
 import (
 	"assignment-exec/image-builder/builder"
 	"assignment-exec/image-builder/configurations"
+	"assignment-exec/image-builder/environment"
+	"flag"
 	"log"
+	"os"
 )
 
+var pushImage = flag.Bool("pushImage", false, "Push image to docker hub")
+
+func init() {
+	flag.BoolVar(pushImage, "p", false, "Push image to docker hub")
+}
+
 func main() {
+	flag.Parse()
 	log.Println("Creating Dockerfile...")
 
+	// Setting environment variables
+	err := os.Setenv("CODE_RUNNER_YAML","code-runner.yaml")
+	if err != nil {
+		log.Fatalf("error while setting environment variables: %v", err)
+		return
+	}
+	os.Setenv("DOCKER_AUTH_YAML", "docker-auth.yaml")
+	if err != nil {
+		log.Fatalf("error while setting environment variables: %v", err)
+		return
+	}
+	os.Setenv("DOCKERFILE_PATH", "Dockerfile")
+	if err != nil {
+		log.Fatalf("error while setting environment variables: %v", err)
+		return
+	}
+	os.Setenv("DOCKER_IO_PATH", "docker.io")
+	if err != nil {
+		log.Fatalf("error while setting environment variables: %v", err)
+		return
+	}
```

Comment:
Environment variables are not set this way.
Please refer the following links to get a better understanding of how environment variables are used.
Note that environment variables not used by the application to know the environment that they are running in. For example, Java doesn't decide where JAVA_HOME is pointing. You have to manually set that env variable so that Java knows where JAVA_HOME is located.
Environment variables can also be set when running docker containers, that are used by the application. Some of those environment variables might have default values and some might not. All the required environment variables have to be declared and assigned default values in the Dockerfile.

https://docs.docker.com/develop/develop-images/dockerfile_best-practices/
https://medium.com/@mccode/dont-embed-configuration-or-secrets-in-docker-images-7b2e0f916fdd
https://engineering.bitnami.com/articles/best-practices-writing-a-dockerfile.html

Reply:
Got rid of environment variables